dwave/optimization/generators.py

```diff
@@ -525,10 +525,16 @@ class as the decision variable being optimized, with permutations of its
     time_distance_ext = np.zeros((num_customers + 1, num_customers + 1))
     time_distance_ext[:num_customers, :num_customers] = time_distances_array[1:, 1:]
 
-    time_depo_ext = np.zeros(num_customers + 1)
-    time_depo_ext[:num_customers] = time_distances_array[0, 1:]
+    # Time from depot to destination
+    time_from_depo_ext = np.zeros(num_customers + 1)
+    time_from_depo_ext[:num_customers] = time_distances_array[0, 1:]
 
-    t_depo = model.constant(time_depo_ext)
+    # Time from source to depot
+    time_to_depo_ext = np.zeros(num_customers + 1)
+    time_to_depo_ext[:num_customers] = time_distances_array[1:, 0]
+
+    t_from_depo = model.constant(time_from_depo_ext)
+    t_to_depo = model.constant(time_to_depo_ext)
```

Comment:
I suppose you could do
```python
    if np.equal(time_from_depo_ext, time_to_depo_ext).all():
         t_from_depo, t_to_depo = model.constant(time_from_depo_ext)
    else:
        t_from_depo = model.constant(time_from_depo_ext)
        t_to_depo = model.constant(time_to_depo_ext)
```
but that's probably premature optimization.

Reply:
Doesn't hurt to add if it's just a couple lines of difference